src/main/webapp/app/exercises/programming/shared/utils/programming-exercise.utils.ts

```diff
@@ -24,27 +25,42 @@ export const isLegacyResult = (result: Result) => {
  * - The submission date of the result / result completionDate is before the buildAndTestAfterDueDate
  *
  * Note: We check some error cases in this method as a null value for the given parameters, because the clients using this method might unwillingly provide them (result component).
- * TODO: Remove the null checks when the result component is refactored.
  *
- * @param result Result with attached Submission - if submission is null, method will use the result completionDate as a reference.
+ * @param latestResult Result with attached Submission - if submission is null, method will use the result completionDate as a reference.
  * @param programmingExercise ProgrammingExercise
  */
-export const isResultPreliminary = (result: Result, programmingExercise: ProgrammingExercise | null) => {
+export const isResultPreliminary = (latestResult: Result, programmingExercise?: ProgrammingExercise) => {
     if (!programmingExercise) {
         return false;
     }
-    const { submission } = result;
-    // We use the result completionDate as a fallback when the submissionDate is not available (edge case, every result should have a submission).
-    let referenceDate = submission && submission.submissionDate ? submission.submissionDate : result.completionDate;
+    let resultCompletionDate = latestResult.completionDate;
+    // We use the result completion date
+    if (!resultCompletionDate) {
+        // in the unlikely case the completion date is not set yet (this should not happen), it is preliminary
```

Comment:
Are you sure that this _should_ not happen? Locked results are always without a completion date. Or is it ensured in advance that the passed result has no completion date?

Reply:
those result are not shown to students, and are (normally) filtered on the server.
In any case it would mean the result is preliminary (because in case it is manual it was not submitted yet), therefore we return true. Automatic results always have a completion date